core/src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -126,6 +128,10 @@ public DateFieldMapper build(BuilderContext context) {
 
         protected void setupFieldType(BuilderContext context) {
             FormatDateTimeFormatter dateTimeFormatter = fieldType().dateTimeFormatter;
+            if (fieldType().dateTimeFormatter().equals(Defaults.DATE_TIME_FORMATTER)) {
```

Comment:
What if the user explicitly sets to the default? I think the correct thing to do is change the initial value in the field type to null, and then add an assert in freeze (override) to make sure it was setup (non null)?